app/static/js/createEvents.js

```diff
@@ -173,48 +184,91 @@ $('#multipleOfferingSave').on('click', function() {
     }
   }
 
+  // Add past date validation
+  datePickerInputs.each(function(index, element) {
+    if (!allowPastStart && isDateInPast($(element).val())) {
+      $(element).addClass('border-red');
+      hasPastDates = true;
+    } else {
+      $(element).removeClass('border-red');
+    }
+  }); 
+
   if (isEmpty){
     let emptyFieldMessage = "Event name or date field is empty";
     displayNotification(emptyFieldMessage);
   }
-  else if(!hasValidTimes){
+  else if (!hasValidTimes) {
     let invalidTimeMessage = "Event end time must be after start time";
     displayNotification(invalidTimeMessage);
   }
-  else if (hasDuplicateListings){
+  else if (hasDuplicateListings) {
     let eventConflictMessage = "Event listings cannot have the same event name, date, and start time";
     displayNotification(eventConflictMessage);
   }
+  else if (hasPastDates) {
+    showModalFlashMessage ("Some events have dates in the past. Please correct them or enable 'Allow start date to be in the past'.", "danger");
+  }
   else {
-    saveOfferingsFromModal();
+    let offerings = [];
+    eventOfferings.each(function(index, element) {
+      offerings.push({
+        eventName: $(element).find('.multipleOfferingNameField').val(),
+        eventDate: $(element).find('.multipleOfferingDatePicker').val(),
+        startTime: $(element).find('.multipleOfferingStartTime').val(),
+        endTime: $(element).find('.multipleOfferingEndTime').val()
+      });
+    });
+
+    let offeringsJson = JSON.stringify(offerings);
+    $("#multipleOfferingData").val(offeringsJson);
+
     updateOfferingsTable();
     pendingmultipleEvents = [];
     $("#checkIsMultipleOffering").prop('checked', true);
-    // Remove the modal and overlay from the DOM
     $('#modalMultipleOffering').modal('hide');
-    $('.invalidFeedback').css('display', 'none');
-    $('#textNotifierPadding').removeClass('pt-5');
     msgFlash("Multiple time offering events saved!", "success");
   }
 });
-
+function showModalFlashMessage(message, type) {
```

Comment:
While I actually think this is a good way to do it, is there a reason you didn't use our built in msgFlash function?

Reply:
When I use the built in msgFlash function, the flash message appears behind the multiple time offering window so the user won't be able to see it.

Reply:
It would be good to add a comment, so that next time, we understand why we needed the extra function.